consai_game/consai_game/tactic/kick/shoot.py

```diff
@@ -0,0 +1,129 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from consai_game.world_model.world_model import WorldModel
+from consai_game.core.tactic.tactic_base import TacticBase, TacticState
+from consai_msgs.msg import MotionCommand
+from consai_msgs.msg import State2D
+from consai_tools.geometry import geometry_tools as tool
+import numpy as np
+
+from transitions import Machine
+
+BALL_NEAR_THRESHOLD = 0.5  # ボールが近いとみなす距離の閾値[m]
+SHOOT_ANGLE_THRESHOLD = 10  # シュート角度の閾値[degree]
+
+
+class ShootStateMachine(Machine):
+    """シュート状態遷移マシン."""
+
+    def __init__(self, name):
+        self.name = name
+        self.move_pos = State2D()
+
+        # 状態定義
+        states = ["chasing", "aiming", "shooting"]
+
+        # 遷移定義
+        transitions = [
+            {"trigger": "ball_near", "source": "chasing", "dest": "aiming"},
+            {"trigger": "ball_far", "source": "aiming", "dest": "chasing"},
+            {"trigger": "shoot", "source": "aiming", "dest": "shooting"},
+            {"trigger": "done_shooting", "source": "shooting", "dest": "chasing"},
+            {"trigger": "reset", "source": "*", "dest": "chasing"},
+        ]
+
+        # ステートマシン構築
+        self.machine = Machine(
+            model=self, states=states, transitions=transitions, initial="chasing"
+        )
+
+    def on_enter_chasing(self):
+        print(f"{self.name} is now chasing the ball.")
+
+    def on_enter_aiming(self):
+        print(f"{self.name} is now aiming at the ball.")
+
+    def on_enter_shooting(self):
+        print(f"{self.name} is shooting!")
+
+
+class Shoot(TacticBase):
+    """指定した位置にシュートするTactic."""
+
+    def __init__(self, target_x=6.0, target_y=0.0):
+        super().__init__()
+        self.target_pos = State2D()
+        self.target_pos.x = target_x
+        self.target_pos.y = target_y
+        self.move_pos = State2D()
+
+        self.robot = ShootStateMachine("robot")
+
+    def reset(self, robot_id: int) -> None:
+        self.robot_id = robot_id
+        self.state = TacticState.RUNNING
+        self.robot.reset()
+
+    def run(self, world_model: WorldModel) -> MotionCommand:
+        command = MotionCommand()
+        command.robot_id = self.robot_id
+        command.mode = MotionCommand.MODE_NAVI
+
+        # ボールの位置を取得
+        ball_pos = world_model.ball.pos
+        robot_pos = world_model.robots.our_robots.get(self.robot_id).pos
+
+        # ロボットとボールの距離を計算
+        dist_to_ball = tool.get_distance(ball_pos, robot_pos)
+
+        # シュートの角度を計算
+        shoot_angle = tool.get_angle(ball_pos, self.target_pos)
+
+        if self.robot.state == "chasing" and dist_to_ball <= BALL_NEAR_THRESHOLD:
+            # ボールが近く、状態がchasingの場合、シュートの準備をする
+            self.robot.ball_near()
```

Comment:
Tactic側で状態遷移を処理しているのが気になります。
状態遷移の責務はMachineに渡しちゃいましょう。

ShootStateMachineに、update()関数とdist_to_ball, shoot_angleメンバーを追加すれば、状態遷移を完全に移植できます

```suggestion
        self.robot.dist_to_ball = tool.get_distance(ball_pos, robot_pos)
        self.robot.shoot_angle = tool.get_angle(ball_pos, self.target_pos)
        self.robot.update()
        if self.robot.state == "chasing":
            # ボールが近く、状態がchasingの場合、シュートの準備をする
```

Reply:
Tacticでやることは、状態に対する"行動" だけにしましょう〜